### DISCUSSION TITLE
Skip re-runs?

### DISCUSSION THREAD

**Discussion by jayjay181818** [Zibbp/ganymede#401] [Q&A] [answered]
I have a streamer enabled that often does 24 hour re-runs while they aren't live, is there a way I can exclude them from being archived? Thanks

**Comment by Zibbp** [comment]
Are these being archived via live streams or VODs? Can you provide an example channel or video ID so I can take a look if there's a field that denotes if it's a rerun or not?

**Comment by jayjay181818** [comment]
Thanks for the prompt response! I do edit for this channel and can confirm that the re-run option is ticked just to confirm.

I only have enabled live streams, Here's an example of a stream it is currently archiving:
ID: ca515079-7962-488e-9a8a-cb5d5d73532e	External ID: 40515879269

**Reply by Zibbp** [reply]
The external ID is a temporary live stream ID that I can't do much with. Can you provide the specific channel so I can take a look?

It looks like the rerun feature just adds a tag to the stream
> Selecting the Rerun checkbox adds a Rerun tag to your stream. Failing to properly label your stream as a Rerun can damage community trust. Without the Rerun label, your viewers expect to engage with a live streamer. Streamers who are properly broadcasting a Rerun will display a Rerun label on their channel page.
https://help.twitch.tv/s/article/video-on-demand?language=en_US#optionsforvideos

If that's the case, I can probably add tags to exclude. A different (better?) option might be to allow a user-customizable title regex filter. If the title includes "rerun", regex can be used to match it and ignore.

**Reply by jayjay181818** [reply]
Tag exclusion would be the ideal solution as the streamer needs to select it if they are running a Rerun. The only thing with this, checking the tags of my friend's stream, it doesn't appear to add it to the tags under the title, just checked my editor viewer however and it is indeed checked. I'm wondering if the title route might be the only way about it..

The username for the streamer I am testing this with is "xtasiaego"

**Reply by Zibbp** [reply]
I've merged https://github.com/Zibbp/ganymede/pull/409 into `main` which adds support for filtering titles with regex. These can be applied to live streams or videos (vod) downloads. If you want to give it a try, set your API and frontend image tags to use `:main`. As this is pretty complex I've created a wiki page with more information: https://github.com/Zibbp/ganymede/wiki/Watched-Channel-Title-Regex. On the page is an example for excluding reruns which should work for your original issue. Let me know how this works for you!.

**Comment by jayjay181818** [comment]
Thanks! I've just added that to one of my pi's to test before I deploy it to my other two docker hosts.
I presume that it's case sensitive so I added a few variations to test with.
So currently I have Negative ticked as well as apply to video downloads, the current title Regex: "(?i:rerun|RERUN|Rerun|[Rerun]|[RERUN]|[rerun])"

**Reply by Zibbp** [reply]
That looks right. You can simplify the regex as the `i` makes it case insensitive. See https://regex101.com/r/WApLox/1 for an example. If you want to be more strict and only match if rerun is in brackets this should do https://regex101.com/r/9fO3FQ/1

**Comment by Zibbp** [comment, marked as answer]
Released in [v2.3.0](https://github.com/Zibbp/ganymede/releases/tag/v2.3.0)
